### PR TITLE
Change `make test` to run outside docker (close #1929)

### LINKED ISSUE #1929: Change `make test` to run the tests from outside the docker container
Right now it's weird to have to either do `make make ARGS=test` or do `docker shell` and _then_ run the tests inside the container (see #1925).

We should change `make test` to run things from the user's working directory.

### PR DESCRIPTION
**Pull request by tofumatt**
r? @EnTeQuAk and @mstriemer (and/or other people who know more about `make` than me)

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
Looks great to me.

**Comment by tofumatt** [comment]
@EnTeQuAk Yeah, I have #1925 as they're already out-of-date, but I'll actually add them to this PR and do them at the same time. I'll take the `pull-request-ready` off this for now.

**Comment by EnTeQuAk** [comment]
Awesome!

**Comment by tofumatt** [comment]
@EnTeQuAk I added docs; r?

File: docs/topics/development/tests.rst [hunk 1]
```diff
@@ -2,25 +2,41 @@
...
+    make test
 
-There's also a few useful makefile targets like ``test``, ``tdd`` and
-``test_force_db``::
+Other, more niche, test commands::
 
-    make test
+    make test_failed # rerun the failed tests from the previous run
+    make test_force_db # run the entire test suite with a new database
+    make tdd # run the entire test suite, but stop on the first error
```
Comment (EnTeQuAk) [inline docs/topics/development/tests.rst]: and starts up pdb

Comment (tofumatt) [inline docs/topics/development/tests.rst]: I copied this from the `Makefile` and it didn't mention that. Do we need to here?

Comment (EnTeQuAk) [inline docs/topics/development/tests.rst]: I don't think we do, it would also make the comment too long.

**Comment by EnTeQuAk** [comment]
r+ :beers:

**Pull request merged**